Sources/ManagePermissions.php

```diff
@@ -2340,30 +2355,19 @@ function loadIllegalBBCHtmlGroups()
 {
 	global $context, $smcFunc;
 
-	if (empty($context['permissions_excluded']['bbc_html']))
-		$context['permissions_excluded']['bbc_html'] = array(-1, 0);
-	else
-	{
-		// Just because you're paranoid doesn't mean they aren't after you.
-		$context['permissions_excluded']['bbc_html'] = array_filter((array) $context['permissions_excluded']['bbc_html'], function ($v)
-			{
-				return is_int($v) || is_string($v) && (string) intval($v) === $v;
-			});
-
-		$context['permissions_excluded']['bbc_html'] = array_unique(array_merge(array(-1, 0), $context['permissions_excluded']['bbc_html']));
-	}
+	$context['permissions_excluded']['bbc_html'] = array(-1, 0);
 
 	$minimum_permissions = array('admin_forum', 'manage_membergroups', 'manage_permissions');
 
 	$request = $smcFunc['db_query']('', '
 		SELECT id_group
 		FROM {db_prefix}membergroups
-		WHERE id_group NOT IN (1, COALESCE((
+		WHERE id_group != 1 AND id_group NOT IN (
```

Comment:
Did `id_group` `1` in the `NOT IN` not work?

Reply:
Nope that doesn't work,
you can't combine a single value and a subquery (which return more than one value) in on ``IN``.

Reply:
The way I had it before was creating an array with a integer and then a sub-array. In PHP the equivalent would have been `array(1, array(2,4,5,6))` when what we wanted was `array(1,2,4,5,6)`. Maybe I could have mucked about some more with `COALESCE` and the sub-query to get the same effect in the end, but this works cleanly.

Reply:
Alternative solution would be:
```
SELECT id_group
FROM {db_prefix}membergroups
WHERE id_group NOT IN (1, COALESCE((
WHERE id_group NOT IN (
	SELECT DISTINCT id_group
	FROM {db_prefix}permissions
	WHERE permission IN ({array_string:permissions})
		AND add_deny = {int:add}
	UNION
	SELECT 1
)
```
but i never see this style of sql command in smf